ui/dashboards/src/context/DashboardProvider.tsx

```diff
@@ -0,0 +1,93 @@
+// Copyright 2022 The Perses Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import create from 'zustand';
+import type { StoreApi } from 'zustand';
+import createZustandContext from 'zustand/context';
+import produce from 'immer';
+import { DashboardSpec, LayoutDefinition, PanelDefinition } from '@perses-dev/core';
+
+interface DashboardState {
```

Comment:
For my own reference, are you thinking that this is where we will store most global dashboard editing state? For example, when we add template variables and we need to show a modal/drawer to edit those values.

Reply:
correct :)